### PR TITLE
Make Chaos Apps exit on service startup failure.

### PR DESCRIPTION
**Pull request by gkleiman**
If one or more services fail to start, e.g., because another process is
already bound to the HTTP port, then the app will log the error and
exit.

### CODE REVIEW

File: src/main/scala/mesosphere/chaos/App.scala [hunk 1]
```diff
@@ -29,23 +31,34 @@ trait App extends scala.App {
...
     sys.addShutdownHook(shutdownAndWait())
 
-    classes.foreach(c => {
-      val service = injector.getInstance(c)
-      services += service
-      service.startAsync()
-    })
+    serviceManager.startAsync()
+
+    Try { serviceManager.awaitHealthy() } match {
```
Comment (kolloch) [inline src/main/scala/mesosphere/chaos/App.scala]: I guess a normal `try` with `catch` is more idiomatic.

File: src/main/scala/mesosphere/chaos/App.scala [hunk 2]
```diff
@@ -29,23 +31,34 @@ trait App extends scala.App {
...
+        log.fatal(s"Failed to start all services. Services by state: ${serviceManager.servicesByState()}", f)
+        shutdownAndWait()
+        System.exit(1)
+    }
   }
 
   def shutdown() {
     log.info("Shutting down services")
-    services.foreach(_.stopAsync())
+    serviceManager.foreach { _.stopAsync() }
```
Comment (kolloch) [inline src/main/scala/mesosphere/chaos/App.scala]: Probably it is just personal taste, so feel free to ignore it but I like parenthesis for small (inline) code blocks without named arguments.

File: src/main/scala/mesosphere/chaos/App.scala [hunk 3]
```diff
@@ -29,23 +31,34 @@ trait App extends scala.App {
...
   }
 
   def shutdownAndWait() {
-    shutdown()
-    log.info("Waiting for services to shut down")
-    services.foreach(_.awaitTerminated())
+    serviceManager.foreach {
```
Comment (kolloch) [inline src/main/scala/mesosphere/chaos/App.scala]: use named argument for multi line function block.

**Comment by kolloch** [comment]
I have some different style preferences, otherwise fine.

**Comment by gkleiman** [comment]
Hope you like my new style ;-)

**Comment by kolloch** [comment]
LGTM!

**Pull request merged**